src/gui/NoteList/NoteList.tsx

```diff
@@ -76,6 +76,15 @@ function NoteList(props: NoteListProps) {
   const [sortDirection, setSortDirection] = useState<SortDirection>(
     props.defaultSortDirection ?? "ascending"
   );
+  const [currentDateFormat, setCurrentDateFormat] = useState("MMM D, YYYY");
+
+  webviewApi
+    .postMessage({
+      type: MsgType.GetJoplinDateFormat,
+    })
+    .then((joplinDateFormat) => {
+      setCurrentDateFormat(joplinDateFormat);
+    });
```

Comment:
For features that are based on a joplin setting, I've adopted a push approach from the plugin back end to the gui. Particularly, it helps keeps the gui in sync whenever the setting changes without the gui having to poll the setting.

Reference Code: https://github.com/rsandz/joplin-calendar/blob/main/src/settings.ts

Could we do the same for the date format? (Same workflow as other settings, but instead of creating our own setting, we use the `getDateFormat` function.)

```mermaid
sequenceDiagram

actor user
participant joplin
participant backend
participant gui

user ->> joplin: Change date format
joplin ->> backend: call setting change callback
backend ->> gui: alert panel that setting changed
```